packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,272 @@
+            ttl: config.subscriptions.managementTokenTTL
+          }
+        );
+      }
+    },
+    {
+      method: 'GET',
+      path: '/subscriptions/plans',
+      options: {
+        auth: {
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        },
+        response: {
+          schema: isA.object().keys({
+            plan_id: isA.string().required(),
+            product_id: isA.string().required(),
+            interval: isA.string().required(),
+            amount: isA.string().required(),
+            currency: isA.string().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listPlans', request);
+        await handleAuth(request.auth);
+        return subhub.listPlans();
+      }
+    },
+    {
+      method: 'GET',
+      path: '/subscriptions/active',
+      options: {
+        auth: {
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        },
+        response: {
+          schema: isA.object().keys({
+            uid: isA.string().required(),
+            subscriptionId: isA.string().required(),
+            productName: isA.string().required(),
+            createdAt: isA.number().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listActive', request);
+        const { uid } = await handleAuth(request.auth);
+        return db.fetchAccountSubscriptions(uid);
+      }
+    },
+    {
+      method: 'POST',
+      path: '/subscriptions/active',
+      options: {
+        auth: {
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        },
+        validate: {
+          payload: {
+            plan_id: isA.string().max(255).required(),
+            token: isA.string().max(255).required()
+          }
+        },
+        response: {
+          schema: isA.object().keys({
+            subscriptionId: isA.string().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.createSubscription', request);
+
+        const { uid, email } = await handleAuth(request.auth);
+
+        await customs.check(request, email, 'createSubscription');
+
+        const planId = request.payload.plan_id;
+        const token = request.payload.token;
+
+        // Find the selected plan and get its product ID
+        const plans = await subhub.listPlans();
+        const selectedPlan = plans.filter(p => p.plan_id === planId)[0];
+        if (! selectedPlan) {
+          throw error.unknownSubscriptionPlan();
+        }
+        const productName = selectedPlan.product_id;
+
+        const paymentResult = await subhub.createSubscription(uid, token, planId);
+
+        const subscriptionId = paymentResult.sub_id;
+        await db.createAccountSubscription({
```

Comment:
The create, update, and delete operations all update an account's profile information and should notify attached devices/services of the profile update. See https://github.com/mozilla/fxa/blob/0d52b0fd6ea8945203e47e4784754e3e8b8b65fc/packages/fxa-auth-server/lib/routes/emails.js#L738:L742

In this case instead of emitting a [`primaryEmailChanged` event](https://github.com/mozilla/fxa/blob/0d52b0fd6ea8945203e47e4784754e3e8b8b65fc/packages/fxa-auth-server/lib/routes/emails.js#L739) to attached services, we should emit a [`profileDataChanged` event](https://github.com/mozilla/fxa/blob/0d52b0fd6ea8945203e47e4784754e3e8b8b65fc/packages/fxa-auth-server/lib/routes/totp.js#L297). Of course, there may be some new subscription events that should be emit instead, maybe @bbangert has some input here.

Reply:
I'll add this general `profileDataChanged` event to to the subscriptions routes.

> there may be some new subscription events that should be emit instead

I think this is basically issue #701 though